velox/common/base/BitUtil.h

```diff
@@ -345,19 +346,35 @@ void forEachBit(
         if (!word) {
           return;
         }
-        while (word) {
-          func(idx * 64 + __builtin_ctzll(word));
-          word &= word - 1;
+        if ((kAllSet & mask) == (word & mask)) {
```

Comment:
Is this optimization important? Do you see a difference in the benchmark results?

The logic seems to rely on mask representing contiguous bits. If so, perhaps, it would be clearer to use __lzcnt64 intrinsic (count number of leading zeros) to calculate "last". __builtin_popcountll calculates number of bits set and gives an impression that random bits might be set. 

It would be nice to add a comment to clarify that mask is expected to represent contiguous set of bits.